packages/tpetra/core/src/Tpetra_Details_unpackCrsMatrixAndCombine_def.hpp

```diff
@@ -340,7 +447,7 @@ struct UnpackCrsMatrixAndCombineFunctor {
     // is an integer in [0, tokens.size()).  It is used to grab a unique (to
     // this thread) subview of the scratch arrays.
     const size_type token = tokens.acquire();
-    const size_t a = static_cast<size_t>(token) * max_num_ent;
+    const size_t a = static_cast<size_t>(token) * batch_size;
```

Comment:
So, now the flat unpack also use batch size as a parameter. What would happen if we run this flat unpack code where the batch size is 1. It will have some functional overhead but mapping of threads to workitems are almost same as the hierarchical version. Just wondering. I vote for removing flat unpacking.